api/envoy/api/v2/listener/listener.proto

```diff
@@ -204,7 +204,18 @@ message FilterChain {
   string name = 7;
 }
 
+// Specifies the match criteria. All of the criteria needs to be satisfied.
```

Comment:
@jplevyak can you take a look at this PR from the PoV of the new Wasm listener filter API you are thinking about? I'd like to make sure that any structural changes we make here to the listener filter model are aligning us.